lib/components/map/gtfs-rt-vehicle-overlay.js

```diff
@@ -0,0 +1,218 @@
+ * @param routes Optional GTFS-like routes list.
+ */
+function gtfsRtToTransitVehicle (gtfsRtVehicle, routes) {
+  const { position, stopId, timestamp, trip, vehicle } = gtfsRtVehicle.vehicle
+  let routeShortName
+  if (routes.length !== 0) {
+    routeShortName = routes.find(r => r.route_id === trip.routeId)?.route_short_name
+  }
+  return {
+    heading: position.bearing,
+    id: vehicle.id,
+    lat: position.latitude,
+    lon: position.longitude,
+    reportDate: new Date(parseInt(timestamp, 10)).toLocaleString(),
+    routeShortName,
+    routeType: 'BUS',
+    stopId,
+    tripId: trip.tripId,
+    vehicleId: vehicle.id
+  }
+}
+
+/**
+ * Generate a vehicle shape component (a colored dot) based on the provided routes list.
+ */
+function makeVehicleShape (routes) {
+  return props => {
+    const { lat, lon, routeShortName } = props.vehicle
+    const route = routes.find(r => r.route_short_name === routeShortName) || {}
+    const iconHtml = ReactDOMServer.renderToStaticMarkup(
+      <VehicleShape route={route}>
+        {routeShortName}
+      </VehicleShape>
+    )
+    return (
+      <Marker
+        icon={L.divIcon({ html: iconHtml })}
+        position={[lat, lon]}
+      />
+    )
+  }
+}
+
+/**
+ * Class that loads and renders a GTFS-rt vehicle positions feed
+ * and associated route information.
+ */
+class GtfsRtVehicleOverlay extends MapLayer {
+  static propTypes = {
+    /** URL to GTFS-rt feed in protocol buffer format. */
+    liveFeedUrl: PropTypes.string,
+    /** URL to GTFS-like route list in JSON format. */
+    routeDefinitionUrl: PropTypes.string
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      routes: [],
+      vehicleLocations: [],
+      // Set to undefined to fall back on the default symbols,
+      // unless a route definition is provided.
+      vehicleSymbols: undefined,
+      visible: props.visible
+    }
+  }
+
+  componentDidMount () {
+    const { liveFeedUrl, name, registerOverlay, routeDefinitionUrl } = this.props
+    registerOverlay(this)
+    if (!liveFeedUrl) {
+      console.warn(`liveFeedUrl prop is missing for overlay '${name}'.`)
+    }
+    if (!routeDefinitionUrl) {
+      console.warn(`routeDefinitionUrl prop is missing for overlay '${name}'.`)
+    } else {
+      this._fetchRoutes()
+    }
+  }
+
+  componentWillUnmount () {
+    this._stopRefreshing()
+  }
+
+  onOverlayAdded = () => {
+    this.setState({ visible: true })
+    this._startRefreshing()
+  }
+
+  onOverlayRemoved = () => {
+    this.setState({ visible: false })
+    this._stopRefreshing()
+  }
+
+  createLeafletElement () {}
+
+  updateLeafletElement () {}
+
+  /**
+   * Fetches GTFS-rt vehicle positions (protocol buffer) and converts to
+   * OTP-UI transitVehicleType before saving to component state.
+   */
+  _fetchVehiclePositions = async () => {
+    if (this.props.liveFeedUrl) {
+      try {
+        const response = await fetch(this.props.liveFeedUrl)
```

Comment:
```suggestion
    const { liveFeedUrl } = this.props
    if (liveFeedUrl) {
      try {
        const response = await fetch(liveFeedUrl)
```

Reply:
This and suggestion below updated in https://github.com/opentripplanner/otp-react-redux/pull/401/commits/4348cdb2c20d2658d9b4f113dd15e18d3b110939.